curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/66c395a16c3ff33dc6f8e7c0.md

```diff
@@ -0,0 +1,133 @@
+---
+id: 66c395a16c3ff33dc6f8e7c0
+title: Step 26
+challengeType: 0
+dashedName: step-26
+---
+
+# --description--
+
+The `.sort()` method mutates the original array - in other words, it modifies the order of the elements directly. This is generally considered bad practise, as it can result in unexpected side effects.
+
+Instead, you should use the `.toSorted()` method, which creates a new array. Change your `.sort()` call to `.toSorted()`. Do not modify the callback function.
+
+# --hints--
+
+You should use `.toSorted()` instead of `.sort()`.
+
+```js
+assert.notMatch(getMedian.toString(), /array\.sort/);
+assert.match(getMedian.toString(), /array\.toSorted/);
+```
+
+You should not modify the callback function.
+
+```js
+assert.match(getMedian.toString(), /array.sort\(\s*\(\s*a,\s*b\s*\)\s*=>\s*a\s*-\s*b\);?/
```

Comment:
```suggestion
assert.match(code, /array\.toSorted\(\s*\(\s*a,\s*b\s*\)\s*=>\s*a\s*-\s*b\);?/);
```

1. The match method was never closed correctly.
2. It doesn't match using `getMedian.toString()` because the code is transformed.


```
console.log(getMedian.toString())
```

Returns:

```
function getMedian(array) {
  if (_LPC2++ % 2000 === 0 && Date.now() - _LP2 > 1500) {
    (function (line) {
      console.log("Potential infinite loop detected on line " + line + ". Tests may be failing because of this.");
    })(3, 0);
    return;
  }
  var sorted = array.toSorted(function (a, b) {
    return a - b;
  });
}
```

---

As an aside, you can run the test using the title, id, or one of the other methods.

```
FCC_CHALLENGE_ID=66c395a16c3ff33dc6f8e7c0 pnpm run test:curriculum
```

https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#testing-challenges